From 562893e665a6c9e1b60c8b3242bab6fe78318b3b Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Fri, 14 Feb 2020 08:19:54 +0000
Subject: [PATCH] configure.ac: make ndiff depend on python3

Python 2 ceased being maintained on the 1st January 2020.
We've already removed all users of it from oe-core, so
let ndiff depend on python3.

Upstream-Status: Pending

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 .../nmap/7.80-r0/nmap-7.80/configure.ac               | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9d2fff8..5ffdd55 100644
--- a/configure.ac
+++ b/configure.ac
@@ -237,16 +237,21 @@ if test $HAVE_PYTHON && test "x${PYTHON_VERSION%%.*}" = "x2"; then
     HAVE_PYTHON2=true
 fi
 
+HAVE_PYTHON3=false
+if test $HAVE_PYTHON && test "x${PYTHON_VERSION%%.*}" = "x3"; then
+    HAVE_PYTHON3=true
+fi
+
 NDIFFDIR=ndiff
 
 # Do they want Ndiff?
 AC_ARG_WITH(ndiff, AC_HELP_STRING([--without-ndiff], [Skip installation of the Ndiff utility]), [], [with_ndiff=check])
-if $HAVE_PYTHON2 ; then : ;
+if $HAVE_PYTHON3 ; then : ;
 else
   if test "$with_ndiff" = "check" ; then
-    AC_MSG_WARN([Not building Ndiff because Python 2.x with x>=4 was not found])
+    AC_MSG_WARN([Not building Ndiff because Python3 was not found])
   elif test "$with_ndiff" = "yes"; then
-    AC_MSG_FAILURE([--with-ndiff requires Python 2.x with x>=4])
+    AC_MSG_FAILURE([--with-ndiff requires Python3])
   fi
   with_ndiff=no
 fi
-- 
2.24.1

